activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -410,26 +410,27 @@ def schema_up_to_date?(configuration, format = ActiveRecord.schema_format, file
         end
       end
 
-      def reconstruct_from_schema(db_config, format = ActiveRecord.schema_format, file = nil) # :nodoc:
-        file ||= schema_dump_path(db_config, format)
+      def reconstruct_from_schema(db_config, file = nil) # :nodoc:
+        file ||= schema_dump_path(db_config, db_config.format)
 
         check_schema_file(file) if file
 
         with_temporary_pool(db_config, clobber: true) do
-          if schema_up_to_date?(db_config, format, file)
+          if schema_up_to_date?(db_config, file)
```

Comment:
Here you're passing the file as the second positional argument, which is `format`. I think you need to pass a `nil` as the second arg for now, and `file` as the third arg.

Reply:
D'oh, this is what happens when I rush feedback. Thanks.